File: 55 Jump Game/55.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        curMax = 0
        for i in range(len(nums)):
            curMax = max(curMax, i + nums[i]) if i <= curMax else curMax
        return curMax >= len(nums) - 1

File: 55 Jump Game/55.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    if (!nums) {
        return false;
    }
    if (nums.length === 1) {
        return true;
    }
    let curMax = 0, n = nums.length
    for (let i = 0; i < n; ++i) {
        curMax = i <= curMax ? Math.max(curMax, i + nums[i]) : curMax
    }
    return curMax >= n - 1;
};